arroyo/processing/strategies/filter.py

```diff
@@ -29,7 +33,22 @@ def submit(self, message: Message[TPayload]) -> None:
         assert not self.__closed
 
         if self.__test_function(message):
-            self.__next_step.submit(message)
+            cast_message = cast(Message[Union[FilteredPayload, TPayload]], message)
+            self.__next_step.submit(cast_message)
+            self.__consecutive_drop_count = 0
+        else:
+            self.__consecutive_drop_count += 1
+
+            if (
+                self.__consecutive_drop_limit is not None
+                and self.__consecutive_drop_count >= self.__consecutive_drop_limit
+            ):
+                cast_message = cast(
```

Comment:
`message.replace()` returns `Message[FilteredPayload]` which does not fit into a `Message[Union[FilteredPayload, TPayload]]` shaped hole

therefore `Message` is missing covariance